Add arbitrary impl for types used in PSBTs

shinghim opened the pull request.

apoelstra commented:
concept NACK these ones -- we aren't allowed to put arbitrary data into these structures. All the C library guarantees us is that the size is OK to assume. In particular, this will produce invalid x-only public keys which will cause aborts if used with the rest of the library.

shinghim commented:
Oh i see. Would it be acceptable to create random SecretKey from the doc example:

```
/// let secp = Secp256k1::new();
/// let secret_key = SecretKey::new(&mut rand::rng());
```

and use that to generate the arbitrary types in `secp256k1` and leave the ones in `ffi` alone?

shinghim commented:
Updated to derive `XOnlyPublicKey` from an arbitrary `PublicKey` and added an impl for `SecretKey`. I removed the impl for `Signature` since I didn't think there would be any use for an arbitrary one. I think it might more make sense to generate an arbitrary `SecretKey` and use that to create an "arbitrary" `Signature`

shinghim commented:
Updated after running `cargo fmt --all` (I think you meant `fmt` instead of `check`). The missing comma you mentioned was the only thing it added

tcharding reviewed (approved):
ACK 02ac1f216bea85bbef43b419fd0c2f8453e5f7d2

shinghim commented:
> Is there any chance these loops spin forever making debugging fuzzing harder?

I don't think so since we're checking the amount of bytes left each time

tcharding commented:
Ah sweet, I misread that check as checking the fill buffer length. Thanks man.

apoelstra commented:
Heh, interesting. I considered nitting the check with "eh, who cares if it 0-fills" but actually, since all-zeroes is not a valid x coordinate, the zero-filling would've caused an infinite loop! Good save.
